### PR TITLE
Added support for computing invariant CRC over IPv6

### PR DESCRIPTION
**Pull request by salehma**
I wanted to use the compute_icrc function as a part of a task in my job. 
I have added the the IPv6 implementation after it was marked under TODO.
I have read the manual on how to implement it and tested.

### CODE REVIEW

**Comment by guedou** [comment]
Thanks for your interest in Scapy and this PR. Could you add a unit test to the file roce.uts?
It helps reviewing your changes and test them.

File: scapy/contrib/roce.py [hunk 1]
```diff
@@ -157,30 +157,38 @@ def compute_icrc(self, p):
...
+            pshdr.tc = 0xff
+            pshdr.hlim = 0xff
+
         else:
-            # TODO support IPv6
-            warning("The underlayer protocol %s is not supported.",
+            warning(f"The underlayer protocol %s is not supported.",
                     ip and ip.name)
-            return self.pack_icrc(0)
+            return struct.pack("!I", 0 & 0xffffffff)[::-1]
```
Comment (guedou) [inline scapy/contrib/roce.py:177]: Why are you not using `pack_icrc(0)` here?

**Comment by guedou** [comment]
flake8 is reporting linting errors. Can you fix them?

**Comment by gpotter2** [comment]
Closed by #4154.

**Pull request closed without merging**